samples/extensions/debug_utils/debug_utils.cpp

```diff
@@ -0,0 +1,1135 @@
+		sampler.addressModeU        = VK_SAMPLER_ADDRESS_MODE_CLAMP_TO_EDGE;
+		sampler.addressModeV        = sampler.addressModeU;
+		sampler.addressModeW        = sampler.addressModeU;
+		sampler.mipLodBias          = 0.0f;
+		sampler.maxAnisotropy       = 1.0f;
+		sampler.minLod              = 0.0f;
+		sampler.maxLod              = 1.0f;
+		sampler.borderColor         = VK_BORDER_COLOR_FLOAT_OPAQUE_WHITE;
+		VK_CHECK(vkCreateSampler(get_device().get_handle(), &sampler, nullptr, &offscreen.sampler));
+	}
+
+	// Bloom separable filter pass
+	{
+		filter_pass.width  = width;
+		filter_pass.height = height;
+
+		// Color attachments
+
+		// Two color buffers
+		create_attachment(VK_FORMAT_R8G8B8A8_UNORM, VK_IMAGE_USAGE_COLOR_ATTACHMENT_BIT, &filter_pass.color[0]);
+
+		// Set up separate renderpass with references to the colorand depth attachments
+		std::array<VkAttachmentDescription, 1> attachment_descriptions = {};
+
+		// Init attachment properties
+		attachment_descriptions[0].samples        = VK_SAMPLE_COUNT_1_BIT;
+		attachment_descriptions[0].loadOp         = VK_ATTACHMENT_LOAD_OP_CLEAR;
+		attachment_descriptions[0].storeOp        = VK_ATTACHMENT_STORE_OP_STORE;
+		attachment_descriptions[0].stencilLoadOp  = VK_ATTACHMENT_LOAD_OP_DONT_CARE;
+		attachment_descriptions[0].stencilStoreOp = VK_ATTACHMENT_STORE_OP_DONT_CARE;
+		attachment_descriptions[0].initialLayout  = VK_IMAGE_LAYOUT_UNDEFINED;
+		attachment_descriptions[0].finalLayout    = VK_IMAGE_LAYOUT_SHADER_READ_ONLY_OPTIMAL;
+		attachment_descriptions[0].format         = filter_pass.color[0].format;
+
+		std::vector<VkAttachmentReference> color_references;
+		color_references.push_back({0, VK_IMAGE_LAYOUT_COLOR_ATTACHMENT_OPTIMAL});
+
+		VkSubpassDescription subpass = {};
+		subpass.pipelineBindPoint    = VK_PIPELINE_BIND_POINT_GRAPHICS;
+		subpass.pColorAttachments    = color_references.data();
+		subpass.colorAttachmentCount = 1;
+
+		// Use subpass dependencies for attachment layput transitions
+		std::array<VkSubpassDependency, 2> dependencies;
+
+		dependencies[0].srcSubpass      = VK_SUBPASS_EXTERNAL;
+		dependencies[0].dstSubpass      = 0;
+		dependencies[0].srcStageMask    = VK_PIPELINE_STAGE_BOTTOM_OF_PIPE_BIT;
+		dependencies[0].dstStageMask    = VK_PIPELINE_STAGE_COLOR_ATTACHMENT_OUTPUT_BIT;
+		dependencies[0].srcAccessMask   = VK_ACCESS_MEMORY_READ_BIT;
+		dependencies[0].dstAccessMask   = VK_ACCESS_COLOR_ATTACHMENT_READ_BIT | VK_ACCESS_COLOR_ATTACHMENT_WRITE_BIT;
+		dependencies[0].dependencyFlags = VK_DEPENDENCY_BY_REGION_BIT;
+
+		dependencies[1].srcSubpass      = 0;
+		dependencies[1].dstSubpass      = VK_SUBPASS_EXTERNAL;
+		dependencies[1].srcStageMask    = VK_PIPELINE_STAGE_COLOR_ATTACHMENT_OUTPUT_BIT;
+		dependencies[1].dstStageMask    = VK_PIPELINE_STAGE_BOTTOM_OF_PIPE_BIT;
+		dependencies[1].srcAccessMask   = VK_ACCESS_COLOR_ATTACHMENT_READ_BIT | VK_ACCESS_COLOR_ATTACHMENT_WRITE_BIT;
+		dependencies[1].dstAccessMask   = VK_ACCESS_MEMORY_READ_BIT;
```

Comment:
Any non-zero access-mask doesn't make sense for `VK_PIPELINE_STAGE_BOTTOM_OF_PIPE_BIT` -- remove.

Reply:
Correction, the output of this pass feeds into the 3rd pass doesn't it? So `VK_PIPELINE_STAGE_BOTTOM_OF_PIPE_BIT` -> `VK_PIPELINE_STAGE_FRAGMENT_SHADER_BIT`, but keep `VK_ACCESS_MEMORY_READ_BIT`?